validator/validator.bal

```diff
@@ -62,3 +131,172 @@ public isolated function validate(json|anydata data, typedesc<anydata>? targetFH
+            if value !is () {
+                errors.push(string `Missing required field '${value.substring()}'`);
+            }
+        }
+
+        //Parsing for missing elements(if resourcetype is msiisng)
+        regexp:Groups? missingElementsData = re `missing required element: "([^""]+)"`.findGroups(data[i]);
+        if missingElementsData is regexp:Groups {
+            regexp:Span? value = missingElementsData[1];
+            if value !is () {
+                errors.push(string `Missing required Element: '${value.substring()}'`);
+            }
+        }
+
+        //Parsing for Invalid fields
+        regexp:Groups? invalidFieldData = re `value of field '([^']+)'`.findGroups(data[i]);
+        if invalidFieldData is regexp:Groups {
+            string fieldName = "";
+            string fieldData = "";
+            regexp:Span? value = invalidFieldData[1];
+            if value !is () {
+                fieldName = string `Invalid field '${value.substring()}'`;
+            }
+            //To get the expected type from the error message
+            regexp:Groups? expectedDataFormat = re `should be of type '([^']+)'`.findGroups(data[i]);
+            if expectedDataFormat is regexp:Groups {
+                regexp:Span? dataType = expectedDataFormat[1];
+                if dataType !is () {
+                    fieldData = string `Type of field should be '${dataType.substring()}'`;
+                }
+            }
+            errors.push(string `${fieldName}. ${fieldData}`);
+        }
+
+        //Parsing for invalid field values
+        regexp:Groups? invalidValuesData = re `^\s*field '([^']+)'`.findGroups(data[i]);
+        if invalidValuesData is regexp:Groups {
+            string valueName = "";
+            string valueData = "";
+            regexp:Span? value = invalidValuesData[1];
+            if value !is () {
+                valueName = string `Invalid value of field '${value.substring()}'`;
+            }
+            //To get the expected type from the error message
+            regexp:Groups? expectedDataFormat = re `should be of type '([^']+)'`.findGroups(data[i]);
+            if expectedDataFormat is regexp:Groups {
+                regexp:Span? dataType = expectedDataFormat[1];
+                if dataType !is () {
+                    valueData = string `Type of value should be '${dataType.substring()}'`;
+                }
+            }
+            errors.push(string `${valueName}. ${valueData}`);
+        }
+
+        //Parsing for invalid array elements
+        regexp:Groups? invalidArrayElementsData = re `^\s*array element '([^']+)'`.findGroups(data[i]);
+        if invalidArrayElementsData is regexp:Groups {
+            string valueName = "";
+            string valueData = "";
+            regexp:Span? value = invalidArrayElementsData[1];
+            if value !is () {
+                valueName = string `Invalid array element '${value.substring()}'`;
```

Comment:
Why do we need a substring here? can't just use ${value} ?

Reply:
Cannot use ${value} because value is of type regexp:Span which returns a list of the data from the regular expression match.